### PR TITLE
test: precedence of flag vs toml for abci client type

### PR DESCRIPTION
**Pull request by JimLarson**
## Description

Refs: Agoric/agoric-sdk#9224

Add unit tests to confirm order of precedence for flag and TOML settings for ABCI client type. The previous test in `start_test.go` confirms that the setting is interpreted as desired.

Also more documentation for TOML template. (Flag help strings are more terse and needn't be as wordy.)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by JimLarson** [comment]
Failing lint check expected, since upstream also fails this check.

File: CHANGELOG-Agoric.md
Comment (gibson042) [inline CHANGELOG-Agoric.md:1]: Nice catches here.

File: server/config/toml.go [hunk 1]
```diff
@@ -83,7 +83,9 @@ iavl-disable-fastnode = {{ .BaseConfig.IAVLDisableFastNode }}
 iavl-lazy-loading = {{ .BaseConfig.IAVLLazyLoading }}
 
 # ABCIClientType selects the type of ABCI client.
-# Default is "committing".
+# Valid settings are "committing" (default) or "local".
+# The committing client allows greater query parallelism,
+# but the local client is more defensive.
 abci-client-type = "{{ .BaseConfig.ABCIClientType }}"
```
Comment (gibson042) [inline server/config/toml.go:89]: Ugh, the buggy manual synchronization in this file is painful. I'm torn on whether or not we want to fix "ABCIClientType selects the type of ABCI client" to "abci-client-type selects the type of ABCI client" for correctness or leave it alone for local consistency, but am inclined towards the latter. The whole thing is crying out for a reflective approach à la https://go.dev/play/p/W74nda-DQKZ .

No changes needed here, though.

**Review by gibson042 — APPROVED** [review]

File: server/config/toml.go (hunk 1 above)
Comment (gibson042) [reply server/config/toml.go:89]: https://github.com/cosmos/cosmos-sdk/issues/20097#issuecomment-2066027842
> Yes 100% agree, we are going to use struct annotations and `github.com/pelletier/go-toml/v2` library for server/v2:
> https://github.com/cosmos/cosmos-sdk/blob/605b724/server/v2/api/grpc/config.go#L19-L34

:tada:

**Pull request merged**